src/lib/Microsoft.Fx.Portability.Reports.DGML/DGMLManager.cs

```diff
@@ -0,0 +1,131 @@
+    /// </summary>
+    class DGMLManager
+    {
+        private readonly Dictionary<string, Guid> _nodesDictionary = new Dictionary<string, Guid>();
+
+        private XElement nodes;
+
+        private XElement links;
+
+        private readonly XNamespace _nameSpace = "http://schemas.microsoft.com/vs/2009/dgml";
+        #region DGML template
+        private readonly string _template =
+            @"<?xml version=""1.0"" encoding=""utf-8""?>
+            <DirectedGraph xmlns=""http://schemas.microsoft.com/vs/2009/dgml"" Background=""grey"">
+            <Nodes>
+            </Nodes>
+            <Links>
+            </Links>
+            <Categories>
+                <Category Id=""VeryHigh"" Background=""#009933"" />
+                <Category Id=""High"" Background=""#ffff66"" />
+                <Category Id=""Medium"" Background=""#ff9900"" />
+                <Category Id=""MediumLow"" Background=""#ff3300"" />
+                <Category Id=""Low"" Background=""#990000"" />
+                <Category Id=""Target"" Background=""white"" />
+                <Category Id=""Unresolved"" Background=""red"" />
+                <Category Id=""Comment"" Label=""Comment"" Description=""Represents a user defined comment on the diagram"" NavigationActionLabel=""Comments"" />
+            </Categories>
+            <Properties>
+                <Property Id=""PortabilityIndex"" Label=""Portability Index"" DataType=""System.String"" />
+            </Properties>
+            <Styles>
+                <Style TargetType=""Node"" GroupLabel=""Comment"" ValueLabel=""Has comment"">
+                  <Condition Expression = ""HasCategory('Comment')"" />
+                  <Setter Property=""Background"" Value=""#FFFFFACD"" />
+                  <Setter Property=""Stroke"" Value=""#FFE5C365"" />
+                  <Setter Property=""StrokeThickness"" Value=""1"" />
+                  <Setter Property=""NodeRadius"" Value=""2"" />
+                  <Setter Property=""MaxWidth"" Value=""250"" />
+                </Style>
+              </Styles>W
+            </DirectedGraph>";
+        #endregion
+
+        XDocument file;
+        public DGMLManager()
+        {
+            file = XDocument.Parse(_template);
+            XElement root = file.Root;
+
+            nodes = root.Element(_nameSpace + "Nodes");
+            links = root.Element(_nameSpace + "Links");
+        }
+
+        public void SetTitle(string title)
+        {
+            file.Root.SetAttributeValue("Title", title);
+        }
+
+        public bool TryGetId(string value, out Guid frameworkGuid)
+        {
+            return _nodesDictionary.TryGetValue(value, out frameworkGuid);
+        }
+
+        internal void AddLink(Guid source, Guid target, string category = null)
+        {
+            var element = new XElement(_nameSpace + "Link",
+                new XAttribute("Source", source),
+                new XAttribute("Target", target));
+
+            if (category != null)
```

Comment:
It's preferred to use `{ }` even with single line if statements. And same for the others below.
```csharp
if (category != null)
{
   element.SetAttributeValue("Category", category);
}
```

Reply:
I agree!